### PR TITLE
Adding tests and support for externalId usage with STS assumeRole

### PR DESCRIPTION
**Pull request by paulmech**
- adding some tests for IAM assumeRole supporting externalId
- refactored IAM.ts to encapsulate ``new STS()`` call to support use of aws-sdk-mock
- updated approval test generator to include IAM examples

### CODE REVIEW

File: spec/STS.assumeRole.cannotAssumeWithoutExternalId.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { IAM } from '../src/IAM'
+// import { STS } from 'aws-sdk'
+
+describe("STS#Assume role fails without externalId",() => {
+    afterEach(() => {
+        AWSMock.restore('STS');
+    })
+    it("should fail to assume a role when the externalId is missing",async () => {
+        try {
+            AWSMock.mock('STS','assumeRole',awsMockFailureCallback('test-data/assumeRole-denied.json'))
```
Comment (temyers) [inline spec/STS.assumeRole.cannotAssumeWithoutExternalId.spec.ts]: Are these only testing the mocks?

Not sure I know a better way of testing, just not sure if these tests are adding anything.  

Documentation is useful, but will they catch an error?

File: package.json [hunk 2]
```diff
@@ -34,6 +34,7 @@
   "dependencies": {
     "aws-sdk": "^2.94.0",
     "chai": "^4.1.2",
+    "global": "^4.3.2",
```
Comment (temyers) [inline package.json]: Is this dependency used?

Comment (temyers) [reply package.json]: https://www.npmjs.com/package/global
Should be removed.

Comment (paulmech) [reply package.json]: yes not required. i will remove

File: spec/STS.assumeRole.cannotAssumeWithoutExternalId.spec.ts (hunk 1 above)
Comment (paulmech) [reply spec/STS.assumeRole.cannotAssumeWithoutExternalId.spec.ts]: Yes they are. It doesn't really differentiate between between an externalId being required and simply not being allowed to assume a role. I will remove it

**Pull request merged**